web-src/src/components/Lyrics.vue

```diff
@@ -0,0 +1,188 @@
+<template>
+  <div
+    class="lyrics-wrapper"
+    ref="lyricsWrapper"
+    @touchstart="scrollAbled = false"
+    @touchend="scrollAbled = true"
+    v-on:scroll.passive="startedScroll"
+    v-on:wheel.passive="startedScroll"
+  >
+    <div class="lyrics">
+      <p
+        v-for="(item, key) in lyricsArr"
+        :class="key == lyricIndex && is_sync && 'gradient'"
+      >
+        {{ item[0] }}
+      </p>
+    </div>
+  </div>
+</template>
+
+<script>
+
+export default {
+  name: "lyrics",
+  data() {
+    // Non reactive
+    // Used as a cache to speed up finding the lyric index in the array for the current time
+    this.lastIndex = 0;
+    // Fired upon scrolling, that's disabling the auto scrolling for 5s
+    this.scrollTimer = null;
+    this.lastItemId = -1;
+    // Reactive
+    return {
+      scroll: {},
+      // lineHeight: 42,
+      scrollAbled: true, // stop scroll to element when touch
```

Comment:
Do you mean scroll enabled with this word?

Reply:
It'd `able to scroll` or `canAutoScroll`.

This is used to disable auto scrolling when the user scrolls the lyrics manually for a short period of time.  Without this, a synchronized lyric will scroll by itself and disturb a user wanting to read them by her own.

Reply:
Shouldn't this be variable simply named `autoScroll`?

Reply:
Sure, I will rename it.

Reply:
Done